wagtail/admin/templates/wagtailadmin/generic/index.html

```diff
@@ -16,15 +16,21 @@
             {% endif %}
         {% endblock %}
     {% endfragment %}
-    {% include "wagtailadmin/shared/header.html" with title=page_title subtitle=page_subtitle action_url=header_action_url action_text=header_action_label action_icon=header_action_icon base_actions=base_actions extra_actions=extra_actions icon=header_icon search_url=search_url search_form=search_form search_results_url=index_results_url only %}
+    {% if not breadcrumbs_items %}
+        {% include "wagtailadmin/shared/header.html" with title=page_title subtitle=page_subtitle action_url=header_action_url action_text=header_action_label action_icon=header_action_icon base_actions=base_actions extra_actions=extra_actions icon=header_icon search_url=search_url search_form=search_form search_results_url=index_results_url only %}
+    {% endif %}
 {% endblock %}
 
 {% block listing %}
-    <div class="nice-padding{% if filters %} filterable{% endif %}">
+    <div class="{% if filters and not breadcrumbs_items %} filterable{% endif %}">
         <div id="listing-results">
+            {% if breadcrumbs_items and locale %}
+                {# TODO: temporarily put the locale selector here since we no longer use the legacy header #}
```

Comment:
Yes this needs adjusting. Seems reasonable enough to me as long as we do complete this in time for the release.